test_monitor_loop_gives_up_after_max_retries times out (pre-existing)

## Bug

`tests/regression/test_issue_40_error_swallowing.py::test_monitor_loop_gives_up_after_max_retries` fails with a 30-second pytest timeout.

## Failure

```
FAILED tests/regression/test_issue_40_error_swallowing.py::test_monitor_loop_gives_up_after_max_retries
Failed: Timeout (>30.0s) from pytest-timeout.
```

The test simulates 5 consecutive monitor loop failures with persistent errors. The monitor loop uses exponential backoff (`retry_delay = min(retry_delay * 2, max_retry_delay)`). The accumulated sleep time across all retries exceeds 30 seconds, hitting the pytest timeout before the loop can give up and set `_running = False`.

## Evidence

Confirmed pre-existing on `main` branch before PR #242. Not introduced by that PR.

## Fix

Either reduce `max_retry_delay` and/or `retry_delay` in the test fixture, or mock `asyncio.sleep` in the test to make it instant.

--- a/tests/regression/test_issue_40_error_swallowing.py
+++ b/tests/regression/test_issue_40_error_swallowing.py
@@ -399,7 +399,7 @@ async def test_monitor_loop_gives_up_after_max_retries(tmp_path, caplog):
     queue_mgr = MessageQueueManager(
         session_manager=mock_session_manager,
         db_path=str(tmp_path / "test.db"),
-        config={"timeouts": {"message_queue": {"initial_retry_delay_seconds": 0.001}}},
+        config={},
     )
 
     # Start monitoring first
@@ -411,12 +411,20 @@ async def test_monitor_loop_gives_up_after_max_retries(tmp_path, caplog):
 
     queue_mgr._get_sessions_with_pending = always_fail
 
-    # real retries complete in ~5ms; poll until loop gives up
-    with caplog.at_level("ERROR"):
-        for _ in range(100):
-            if "giving up" in caplog.text:
-                break
-            await asyncio.sleep(0.05)
+    # Mock asyncio.sleep so exponential backoff runs instantly.
+    # We use the real asyncio.sleep(0) inside so the event loop still switches
+    # tasks and the monitor loop can actually progress between test iterations.
+    _real_sleep = asyncio.sleep
+
+    async def instant_sleep(*args, **kwargs):
+        await _real_sleep(0)
+
+    with patch("asyncio.sleep", new=instant_sleep):
+        with caplog.at_level("ERROR"):
+            for _ in range(100):
+                if "giving up" in caplog.text:
+                    break
+                await asyncio.sleep(0.05)
 
     # Stop monitoring
     await queue_mgr.stop()

